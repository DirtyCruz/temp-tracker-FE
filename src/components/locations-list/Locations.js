import React from 'react';
import PropTypes from 'prop-types';
import Location from './Location';

function Locations({ locationsAndTemps }) {
  const locationsList = locationsAndTemps.map(({ location, temperature }) => (
    <li key={location}>
      <Location 
        temperature={temperature}
        location={location}
      />
    </li>
  ));
  return (
    <ul>
      {locationsList}
    </ul>
  );
}

Locations.propTypes = {
  locationsAndTemps: PropTypes.arrayOf(PropTypes.shape({
    location: PropTypes.string.isRequired,
    temperature: PropTypes.number.isRequired
  })).isRequired
};

export default Locations;
