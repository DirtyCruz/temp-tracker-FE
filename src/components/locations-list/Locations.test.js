import React from 'react';
import { shallow } from 'enzyme';
import Locations from './Locations';

describe('Locations component', () => {
  it('renders Locations', () => {
    const wrapper = shallow(<Locations locationsAndTemps={[{ location: 'Portland', temp: 1000 }]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
