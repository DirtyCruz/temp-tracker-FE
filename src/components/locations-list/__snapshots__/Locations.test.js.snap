// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Locations component renders Locations 1`] = `
<ul>
  <li
    key="Portland"
  >
    <Location
      location="Portland"
    />
  </li>
</ul>
`;
